### PR TITLE
Adding external call example

### PR DESCRIPTION
**Pull request by cjk12**

### CODE REVIEW

**Comment by cjk12** [comment]
retest this please

**Comment by cjk12** [comment]
@galbramc, @bqpd thinks this might be because I committed some files before they were in their final form.  Could you take a look?

**Comment by bqpd** [comment]
(specifically, the new output files created by the new examples have changed intentionally, but seem to be causing the failure?)

**Comment by galbramc** [comment]
I know why the files get modified:

```
git diff docs/source/examples/external_gp_output.txt
diff --git a/docs/source/examples/external_gp_output.txt b/docs/source/examples/external_gp_output.txt
index 4bd7886..673c530 100644
--- a/docs/source/examples/external_gp_output.txt
+++ b/docs/source/examples/external_gp_output.txt
@@ -1,5 +1,5 @@
 Beginning signomial solve.
-Solving took 3 GP solves and 0.0186 seconds.
+Solving took 3 GP solves and 0.0114 seconds.
 
 Cost
 ----
```

You should not have execution times included in these example outputs. It breaks the "gpkit" testing formula. @whoburg told me this is part of your testing process to make sure that none of the example files show up as modified when you run "git status" after running the tests.

**Comment by bqpd** [comment]
Yeah, this is great to have! I had a hard time parsing exactly what the diff was in the console output; does it give the specific diff? I only saw

```
error: found modified files!
error: 	modified:   docs/source/examples/external_gp_output.txt
	modified:   docs/source/examples/sin_approx_example_output.txt
```

**Comment by cjk12** [comment]
So how do I fix this (probably directed at @bqpd)?  Is it a verbosity issue in my solve call?

**Comment by galbramc** [comment]
I can change the script to actually show the diff. Don't push anything to this for a bit, I'm going to use it to setup the testing script.

File: docs/source/examples/external_gp.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+objective = y
+
+constraints = [External_Constraint(x, y),
+               x <= np.pi/2.,
+               x >= np.pi/4.,
+              ]
+
+m = Model(objective, constraints)
+sol = m.localsolve()
```
Comment (bqpd) [inline docs/source/examples/external_gp.py]: `verbosity=1`

**Comment by cjk12** [comment]
Sounds good.  Thanks.

File: docs/source/examples/sin_approx_example.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+objective = y
+
+constraints = [y >= x,
+               x <= np.pi/2.,
+               x >= np.pi/4.,
+              ]
+
+m = Model(objective, constraints)
+sol = m.solve()
```
Comment (bqpd) [inline docs/source/examples/sin_approx_example.py]: `verbosity=1`

**Comment by bqpd** [comment]
@cjk12, yup! Commented inline. @galbramc, that would be great, thanks!

**Comment by galbramc** [comment]
test this please

**Comment by cjk12** [comment]
@bqpd Ok, I'll update after @galbramc makes his changes.

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
@bqpd looks good? https://acdl.mit.edu/csi/job/gpkit_PullRequest_Unit/buildnode=reynolds,optimizer=cvxopt/1208/console

**Comment by bqpd** [comment]
Looks great!

**Comment by galbramc** [comment]
Ok @cjk12 , go ahead and push the fixes

**Comment by galbramc** [comment]
@cjk12, you need to push updated output files as well.

**Comment by cjk12** [comment]
Oh...  Oops.  One sec...

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
the tests actually passed, there was a connection problem on the windows virtual machine

**Comment by galbramc** [comment]
How you have to pass the Hoburg review. I can't help you with that one. Best of luck :P

**Comment by cjk12** [comment]
Haha.  Great, thanks for the help @galbramc

**Comment by cjk12** [comment]
@whoburg ready for review

**Comment by galbramc** [comment]
test this please

**Comment by whoburg** [comment]
@cjk12, this is great! I made a couple tweaks. I'm going to rebase this to squash some of the line ending commits and then merge it.

**Comment by whoburg** [comment]
this is ready to merge once it passes tests.

**Pull request merged**